### ISSUE TITLE
Unable to sign-in on Huawei

### ISSUE THREAD

**Issue by Sam101011** [microg/GmsCore#2091]
**Describe the bug**

The latest changelog states:
>Allow sign-in without linking device. Useful for devices that are not qualified to run Google Play Services (e.g. because of Huawei ban).

I've tried it numerous times, cleared data and reinstalled to no avail.

**To Reproduce**
1. Install microG services core and all of the other required components
2. Enable Google account registration
3. Attempt logging in

**Expected behavior**
I hoped I'd finally be able to login as that'd allow me to use the couple gapps I still find useful

**Screenshots**
![Screenshot_20231114_061423_com google android gms](https://github.com/microg/GmsCore/assets/90930406/73a7f58e-e437-40a8-a035-1b4c501a11d0)

**System**
Android Version: 10
Custom ROM: EMUI 12
Device: Huawei p40 lite(JNY-LX1)

**Comment by Aeon1704** [comment]
I have same bug... or whatever it is. Tried Aurora Store with google account after that, it worked. MicroG didn't.

**Comment by Sapiosenses** [comment]
There is a test build of GMScore that is reputedly designed for Huawei devices - though this is not publicly documented.

I have talked to Huawei owners who claim it works, but the microG project officially advises people not to try it unless someone from microG asks them to.

See the reference to the "-hw" builds in the latest release notes:

https://github.com/microg/GmsCore/releases/tag/v0.3.0.233515

**Comment by Sam101011** [comment]
Nope. Neither hw nor lh work. The wait goes on

**Comment by Aeon1704** [comment]
Can confirm. Not working.

**Comment by bulbulogludemir** [comment]
Afaik, the only devices can run the hw version of MicroG is the ones running on EMUI 13.1... So, If your device is below EMUI 13.1, then no chance....

**Comment by Sam101011** [comment]
Bruh

**Comment by bulbulogludemir** [comment]
Can you log in to your google account in YouTube, Google Maps...etc. ?

**Comment by Sam101011** [comment]
Just tried it and it worked
BUT
I can't sign in into any google apps, which makes the whole thing pointless. I still get the pop-ups saying "Play Services are not supported-bla-bla". Trying to login again just shows that the profile was already added. Tried with gAuthenticator, Translate and Gmail.

Uninstalled, reinstallled and tried with all the other profiles as well. Same result every time. And tbh even if I knew how to create custom profiles, it's highly likely that it wouldn't work either. It looks like signature spoofing is still required, despite it being completely impossible to achieve on Huawei.

Idk why the Huawei login option hasn't been added yet, like on Vanced microG, along with dark mode(probably some technical issues).

TL:DR: You can login in microG, but not in any gapps

**Comment by mar-v-in** [comment]
The -hw version is currently only meant for testing on Huawei P60 Pro with latest EMUI version. Before reporting issues, always make sure that none of the items in microG Self-Check has an unchecked box. If you are using a Huawei device in Malaysia or Thailand, you can also try to install the -lh version of microG from AppGallery.

**Comment by SamWang3047** [comment]
I fixed the sign-in problem, try to set your date and time to Auto. Then try to sign in again.

**Comment by SamWang3047** [comment]
I fixed the sign-in problem, it works for me, try to set your date and time to Auto sync. Then try to sign in again.

**Comment by NeoGeo2K** [comment]
Try many versions. Found work for me (MatePad T10S).

https://dl.youtubevanced.org/new/microg/microg-latest-DEV-By-Vanced%20-team-(YouTubeVanced.org).apk

Old version. But work. Have button Huawei on first logon screen.

All other versions didnt have this button and get error "Try later"

**Comment by 3laaHisham** [comment]
> Try many versions. Found work for me (MatePad T10S).
> 
> https://dl.youtubevanced.org/new/microg/microg-latest-DEV-By-Vanced%20-team-(YouTubeVanced.org).apk
> 
> Old version. But work. Have button Huawei on first logon screen.
> 
> All other versions didnt have this button and get error "Try later"

Thanks a lot. This worked for me.

**Comment by Parsaww** [comment]
> > Try many versions. Found work for me (MatePad T10S).
> > https://dl.youtubevanced.org/new/microg/microg-latest-DEV-By-Vanced%20-team-(YouTubeVanced.org).apk
> > Old version. But work. Have button Huawei on first logon screen.
> > All other versions didnt have this button and get error "Try later"
> 
> Thanks a lot. This worked for me.

hi is there any chance you could share the apk on another host? the website seems to be suspended

**Comment by LukisG** [comment]
> > > Try many versions. Found work for me (MatePad T10S).
> > > https://dl.youtubevanced.org/new/microg/microg-latest-DEV-By-Vanced%20-team-(YouTubeVanced.org).apk
> > > Old version. But work. Have button Huawei on first logon screen.
> > > All other versions didnt have this button and get error "Try later"
> > 
> > Thanks a lot. This worked for me.
> 
> hi is there any chance you could share the apk on another host? the website seems to be suspended
> 

Please check the solution above, it's already solved. Tldr you need to fresh install it and set your device to emulated then you can login on latest version of microg

**Comment by LukisG** [comment]
> Can't find it the git issue anywhere that helped me fix this but I made it working. Huawei p40 lite,
> I use latest version with -hw in the name and to make it work:
> 1. Disable network,
> 2. Force stop microg
> 3. Clear cache and temp,
> 4. Open microg
> 5. Select device registration profile (arm64-v8a)
> 6. Turn on internet and disable/enable device registration.
> 7. Wait for ID
> 8. Login to Google. 
> 
> Thank me later :) 

@Parsaww

**Comment by 3laaHisham** [comment]
> > > Try many versions. Found work for me (MatePad T10S).
> > > https://dl.youtubevanced.org/new/microg/microg-latest-DEV-By-Vanced%20-team-(YouTubeVanced.org).apk
> > > Old version. But work. Have button Huawei on first logon screen.
> > > All other versions didnt have this button and get error "Try later"
> > 
> > Thanks a lot. This worked for me.
> 
> hi is there any chance you could share the apk on another host? the website seems to be suspended
> 

https://drive.google.com/file/d/10wJD-yd86_QOWznzOewPGtWI5uh-RmS5/view?usp=drivesdk

**Comment by mar-v-in** [comment]
"Vanced microG" is not microG. It is a third party project that patched microG to remove some security functionality to allow their modded YouTube apps to run. At the same time, they broke 90% of microG's functionaloty. If you're using this "Vanced microG" or similar projects (e.g. "MicroG RE"), this is totally unsupported are. In this case please don't report any issues here as it's well known that most things won't work with it.

microG's support for Huawei devices and EMUI is still very young. The device profile feature (which allows to simulate a different device) allows to login on Huawei devices as was described above. Some versions of EMUI can be used with microG -hw and -lh versions, but again, this is still in development. Once it's working fine, there will be official releases on the website and in Huawei AppGallery.

**Comment by Wojtaz0w** [comment]
Vanced microG is never intended as replacement for microG official. Its sole purpose is to support YouTube (Re) Vanced . Do not install it from unofficial websites like the one ending in .org , only use mMicroG or VancedMicroG if u have YouTube ReVanced, otherwise it is useless!!

**Comment by Parsaww** [comment]
> "Vanced microG" is not microG. It is a third party project that patched microG to remove some security functionality to allow their modded YouTube apps to run. At the same time, they broke 90% of microG's functionaloty. If you're using this "Vanced microG" or similar projects (e.g. "MicroG RE"), this is totally unsupported are. In this case please don't report any issues here as it's well known that most things won't work with it.
> 
> microG's support for Huawei devices and EMUI is still very young. The device profile feature (which allows to simulate a different device) allows to login on Huawei devices as was described above. Some versions of EMUI can be used with microG -hw and -lh versions, but again, this is still in development. Once it's working fine, there will be official releases on the website and in Huawei AppGallery.

Hi yes im aware of that , after some research it seems the Vanced microg version had the huawei button copied from a microg fork made specifically for huaweis and i finally found it , it works very well with google apps and drive seems to work with whatsapp backup, play store still doesn't work but Aurora is better.
Heres the repo https://github.com/jcchikikomori/GmsCoreHuawei
Used one version before the latest release.

**Comment by mar-v-in** [comment]
> What is the difference between the -hw and -lh versions, and which one should I use?

The -hw version uses the [Huawei Differentiated Signature Capability](https://forums.developer.huawei.com/forumPortal/en/topic/0202128603315033024). It's an official feature to support usecases like microG that require to be detected under a different signature. It's a relatively new feature, currently (as far as I know) only available to the P60 Pro with latest EMUI version.

The -lh version uses a hack from Lighthouse, a previous microG fork made for Huawei devices in Malaysia and Thailand, to reach the same functionality. It may work with other Huawei devices, but because it's not an official feature, it may break at any time.

The easiest way to figure out which version is working for you, install it and check the Self-Check in microG settings. If the box behind "System spoofs signature" is checked, you're good to go.

If the box is not checked, only a small subset of microG's functionality will be available:
- Push notifications in some apps
- Google sign-in in some apps

Most third party apps made for use with Google Play Services will continue to not work correctly and show corresponding error messages.

@Parsaww you might be in the state I just described, as the version you link implements neither of the two mechanisms.

**Comment by coolkid3245** [comment]
Using -lh version works for me.

Huawei Mate 30 Pro (bought indonesian version) LIO-L29 
EMUI 12.0.0.365 C301

Installed from appgallery using malaysian region.

**Comment by khalidsbaih** [comment]
![1](https://github.com/microg/GmsCore/assets/165248602/bf5ce845-b97c-42d0-98f8-a832da75f9a8)
apply the settings in the photo & it will work

**Comment by bulbulogludemir** [comment]
What's the device, and EMUI version?

**Comment by DN2048** [comment]
> ![1](https://private-user-images.githubusercontent.com/165248602/317638914-bf5ce845-b97c-42d0-98f8-a832da75f9a8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NjYyNzksIm5iZiI6MTcxMTY2NTk3OSwicGF0aCI6Ii8xNjUyNDg2MDIvMzE3NjM4OTE0LWJmNWNlODQ1LWI5N2MtNDJkMC05OGY4LWE4MzJkYTc1ZjlhOC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMjI0NjE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmE1NjRjYzY1MmVlMWYyM2U0OTJhOWNmNWNlNjA2NTQyMDk4MDQ3ZmZjNTk4OTIyN2E4YmFkNmY1NDI5M2ZjMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XB_jsVDOKF46ntehAKyYfJ126MKCmFlTIzYHxPXURFE) apply the settings in the photo & it will work

Worked for me on my Huawei P40 Pro with the latest Vanced microG. Thanks.

**Comment by Gotoro** [comment]
> Can't find it the git issue anywhere that helped me fix this but I made it working. Huawei p40 lite, I use latest version with -hw in the name and to make it work:
> 
> 1. Disable network,
> 2. Force stop microg
> 3. Clear cache and temp,
> 4. Open microg
> 5. Select device registration profile (arm64-v8a)
> 6. Turn on internet and disable/enable device registration.
> 7. Wait for ID
> 8. Login to Google.
> 
> Thank me later :)

This is the solution

**Comment by bulbulogludemir** [comment]
Vanced microG? Doesn't it a thing for YouTube Vanced?

**Comment by bulbulogludemir** [comment]
@Gotoro can you use Google Apps with your google account? YouTube, Chrome, GMail....etc. ?

**Comment by mar-v-in** [comment]
Please stop talking about (Re)Vanced microG in this venue. It already confuses a lot of people that they took the microG name to create an incompatible software, please don't add on top.

This issue is exclusively about signing in with upstream microG on Huawei devices.

**Comment by LukisG** [comment]
Marvin if you can pin this in any way so it stops being replied here is solution for those who is experiencing issues with loging in on huawei:

1. Disable network,
2. Force stop microg
3. Clear cache and temp,
4. Open microg
5. Select device registration profile (arm64-v8a)
6. Turn on internet and disable/enable device registration.
7. Wait for ID
8. Login to Google.

**Comment by Gotoro** [comment]
@bulbulogludemir hey there. I use yt vanced and brave, both work. gmail requires you to sign in with google, which pulls up microg sign-in menu. Unfortunately, even if you login in it just errors out

**Comment by bulbulogludemir** [comment]
@Gotoro oh okay, thank you for the info. I guess still the only way to use microg fully is if you have a device which has EMUI 13.1 or EMUI 14 since those are the only 2 versions which has spoofing enabled.

**Comment by mar-v-in** [comment]
@LukisG with the settings as mentioned above, you don't need to go the route you propose (which may break apps you previously used with microG)

**Comment by mar-v-in** [comment]
Yes, not all devices support -hw yet, but that's Huawei's doing, nothing I can help with. Doesn't it work using the latest -lh build from AppGallery for you?

**Comment by FloriaLysandru** [comment]
> Marvin if you can pin this in any way so it stops being replied here is solution for those who is experiencing issues with loging in on huawei:
> 
> 1. Disable network,
> 2. Force stop microg
> 3. Clear cache and temp,
> 4. Open microg
> 5. Select device registration profile (arm64-v8a)
> 6. Turn on internet and disable/enable device registration.
> 7. Wait for ID
> 8. Login to Google.

Yea, tried this and it worked

**Comment by mar-v-in** [comment]
@rollingmoai This shouldn't be like this. If you have previously used `-lh` version you very likely shouldn't have been updated to `-hw`.
